exonum-java-binding/common/src/main/java/com/exonum/binding/common/proofs/CheckedProof.java

```diff
@@ -25,6 +25,7 @@
  * If it is valid, the proof contents may be accessed. See {@link CheckedListProof}
  * and {@link CheckedMapProof} for available contents description.
  */
+// todo: Why do we need to represent invalid proofs?
```

Comment:
The flat proof currently does *not* implement UncheckedProof interface, that's why it may and does throw exceptions. I decided exceptions are easier to implement and test. The question is on the original reasoning to represent them this way.

There is a year-old https://jira.bf.local/browse/ECR-2410 to reconsider.